JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -18,44 +18,121 @@
+import java.nio.charset.StandardCharsets;
+
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
 import picocli.CommandLine;
 import picocli.CommandLine.Command;
 import picocli.CommandLine.Option;
 import picocli.CommandLine.Parameters;
 
+import com.vertica.jdbc.VerticaConnection;
+
 @Command(name = "OAuthSampleApp", mixinStandardHelpOptions = true, version = "OAuth sample app 1.0", description = "Connects to a Vertica database using OAuth")
 public class OAuthSampleApp implements Callable<Integer> {
 
-    @Parameters(index = "0", description = "Host name")
-    private String host = "";
-
-    @Parameters(index = "1", description = "Database name")
-    private String dbName = "";
-
-    @Option(names = { "-p", "--port" }, description = "Port")
-    private String port = "5433";
+	private static Properties prop;
+	private static String OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+    private static String OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+   
+	private static Connection vConnection;
+	
+	private static String getConnectionString(String args) {
+		Map<String, String> kv=new HashMap<String, String>();
+        String[] entries = args.split(";");
 
-    @Option(names = { "-a", "--access-token" }, description = "Access token")
-    private String accessToken = "";
-
-    @Option(names = { "-r", "--refresh-token" }, description = "Refresh token")
-    private String refreshToken = "";
+        for (String entry : entries) {
+            String[] pair = entry.split("=");
+            kv.put(pair[0], pair[1]);
+        }
+        return "jdbc:vertica://" + kv.get("Host") + ":" 
+								 + kv.get("Port") + "/" 
+								 + kv.get("Database") + "?user=" 
+								 + kv.get("User") + "&password=" 
+								 + kv.get("Password");
+    }
 
-    @Option(names = { "-s", "--client-secret" }, description = "Client Secret")
-    private String clientSecret = "";
+	private static void SetUpDbForOAuth() throws Exception
+    {
+		String connectionString = getConnectionString(prop.getProperty("ConnectionString"));
+    	vConnection = DriverManager.getConnection(connectionString);
+		System.out.println("SetUpDbForOAuth");
+    	String CONNECTION_STRING = prop.getProperty("ConnectionString");    
+        String USER = prop.getProperty("User");
+        String CLIENT_ID = prop.getProperty("ClientId");
+        String CLIENT_SECRET = prop.getProperty("ClientSecret");
+        String INTROSPECT_URL = prop.getProperty("TokenUrl") + "introspect";
+       	Statement st = vConnection.createStatement(); 
+       	
+		st.execute("DROP USER IF EXISTS " + USER + " CASCADE;");
+		st.execute("DROP AUTHENTICATION IF EXISTS v_oauth CASCADE;");
+		st.execute("CREATE AUTHENTICATION v_oauth METHOD 'oauth' LOCAL;");//HOST '0.0.0.0/0';");//getOAuthHost());
```

Comment:
The comment shouldn't be here or should be explained.

Reply:
Will check and remove the comment.